src/JobPanel/backend/service_base.py

```diff
@@ -214,24 +218,61 @@ def _process_requests(self):
                 data = request['data']
             assert( 'action' in request.keys() )
             answer = self.call_action(request['action'], data)
+
+            # TODO:
+            # catch exceptions, use async_repeater._exception_answer(e) to return
+            # exception result. Use format_exception in _exception_answer instead of just traceback.
+            # For correct result introduce similar formater.
+
             self._repeater.send_answer(request_data.id, answer)
         return
 
+    def _do_work(self):
+        pass
+
+
+    def save_result(self, answer_data):
+        (result_list, result_data) = answer_data
+        result_list.append(result_data)
+
+    def get_connection(self, connection_data):
+        return ClassFactory([ConnectionSSH, ConnectionLocal]).make_instance(connection_data)
+
+
+    #######################################################################################
+    # Methods that implements a request but can also be called directly by the service.
+    #
+
+
     """ Requests. """
-    def request_start_child(self, request_data):
+
+    @LongRequest
+    def request_start_child(self, service_data):
         """
-        Start a new child service with given data.
+        Start a new child service with config given by `service_data`.
+        - Marked as LongRequest so it runs in separated thread as it takes longer time to complete.
+
         TODO:
-        - implemented only in: RootService, BackendService
+        - Root starts Backend in Docker either:
+          - docker have same IP as Root, then we can set connection in service_data
+          - docker have separate IP, then we must create container and get its IP before
+            executing backend in the container
 
-        proxy=self.make_child_proxy()
-        proxy.start_service() ... enqueue the seervice
-        :param request_data:
-        :return: `OK`
+        :param service_data: Service configuration data.
+        :return: STATUS
         """
-        address = request_data['socket_address']
-        self.make_child_proxy(address)
-        return self.answer_ok
+        connection = self.get_connection(service_data.connection)
+        # this method should keep active conncetions in a dict and reuse conncetion to same hosts
```

Comment:
Co bude jako klíč v dictionary? Nebylo by lepší kdyby to byl seznam, stejně se budou muset porovnávat všechny položky každého connection na shodu.